### ISSUE TITLE
Fate/Extra background issues on anything but D3D

### ISSUE THREAD

**Issue by TrunerReisar** [hrydgard/ppsspp#10577]
What happens?
On Windows and Android, using OpenGL during the VN sections produce backgrounds that are misaligned in order. You check door two and the background for door one appears. You go up the stairs and the floor default background appears for a bit.
On Vulkan, the backgrounds are completely black with a white smudge.
These backgrounds, I imagine are low res still images used to represent where you are and not created on the fly (based on the differences between character placements).

What *should* happen?
D3D11 shows the backgrounds properly, in the right order.

What hardware and operating system are you running PPSSPP on? GPU might matter if it's a graphical issue.
Windows 10 PC with a GeForce 850m and Android 4 running on a GPD XD, using PPSSPP Gold purchased from Google Play Store, updated to latest.

**Comment by unknownbrackets** [comment]
Interesting.  Do you have a desktop you could try capturing the issue with on Windows?

https://github.com/hrydgard/ppsspp/wiki/How-to-find-a-graphic-issue-with-the-GE-debugger

I'm wondering if it's using mipmapping or else some sort of blending to use the images.  If you click step prim, you may see it draw the right image, and then draw the wrong image on top.  Whatever the scenario, knowing what it's doing would help a lot to figure out this issue.

To clarify - I'm guessing this doesn't happen with software rendering (it's really slow, but more accurate mostly.)

-[Unknown]

**Comment by TrunerReisar** [comment]
Well I tried and as soon as I clicked Step Frame, PPSSPP crashes (it also crashes if I try to save state, which never happened before). Using the latest nightly. The issue is also inconsistent as heck. If I leave Buffered rendering on, only a quarter of the screen renders. If I set it to skip, the whole screen renders, but the delayed background issue happens. If I set it back to buffered, it works properly. Fortunately, this issue can be seen right from the beginning, so perhaps someone else could try debugging (unless the latest nightly just happens to crash for everyone?) I don't see a log file to check what went wrong during the crash in the PPSSPP folder.

**Comment by unknownbrackets** [comment]
Ah.  #10599 will fix the crash with "skip buffer effects".  Unfortunately, it looks like the GE debugger with skip buffer effects has recently been broken.

Just a reminder: the PSP hardware supported buffers.  "Skip buffer effects" is a hack to improve performance on weaker devices.  Some games *intentionally* use effects to blur the screen or otherwise, and this setting also disables them - but may result in various other graphical issues.

Think of it like a "disable windshield" setting for a car emulator.

Any graphics that are broken in "skip buffer effects" are essentially intentional.  The bugs in your eyes without the windshield are just the setting doing exactly what it was designed to do.  Cuts down on wind resistance a bit, though.

So to summarize:
 * In skip buffer effects, the original described problem happens.
 * In buffered rendering, only a quarter of the screen renders.
 * Switching between modes can convince buffered rendering to temporarily work.

Is that right?  And this behavior is the same on all graphics backends?

I'm afraid I don't actually have this game - I have a large % of the English RPGs for the PSP but I think I classified this as a visual novel and skipped it.

-[Unknown]

**Comment by TrunerReisar** [comment]
What you described happens on OpenGL, I forgot to mention that. After the crashes, I came back without testing it on Vulkan. But yes, switching between buffer types fixed the issue and I imagine it should last until PPSSPP is turned off. Will test more later.

Similarly, if I start the game up with skip buffer effects, the backgrounds will show as black only. Swapping to buffered and then skipping back "fixes" the problem by showing the backgrounds, but delaying them as described in the original post. That is new however, as I did not notice that happening before. But I am not sure what my previous PC version of PPSSPP was, as I replaced it just yesterday to make sure the issue I was having wasn't fixed meanwhile already.

I'll try this on the Android version ASAP.

**Comment by TrunerReisar** [comment]
All right, back with some more testing. On Android, having "Simulate Block Transfer Effects" on fixes the problem entirely. Game renders the backgrounds properly, in the right order. The game does take a massive performance hit, going down to the lower 20s and hardly maintaining 30, if ever. But it does mean that the Android version has a permanent way of making the game work.

On Windows, I had this option on by default and it didn't change anything. So on Windows, the bigger problem is that part of the screen doesn't render until buffered rendering is nudged a bit to work. That said, it works after that, so I guess this issue can be considered solved?

**Comment by unknownbrackets** [comment]
It should just work.

Does it depend on the render resolution setting?  Is it a smaller part of the screen the higher the resolution is set to?

-[Unknown]

**Comment by TrunerReisar** [comment]
Aha!
I had rendering resolution set to 3x. Taking it down to either 1 or 2 fixes the issue completely. In fact, after that, I can set it to any resolution and it works.

**Comment by unknownbrackets** [comment]
Interesting.  So we must have a bug handling some effect at higher render resolutions - probably a remaining scissor issue...

-[Unknown]

**Comment by Gamemulatorer** [comment]
No feedback outdated @TrunerReisar

**Comment by hrydgard** [comment]
If this is still an issue, please provide a [GE frame dump](https://github.com/hrydgard/ppsspp/wiki/How-to-create-a-frame-dump) and re-open. Closing until then
